### ISSUE TITLE
openxlsx::writeDataTable saves empty cells for data.frame columns which contain vectors with NAs in them

### ISSUE THREAD

**Issue by romanzenka** [ycphs/openxlsx#298]
**Describe the bug**
When saving a data.frame whose column is a list containing vectors with NAs with them, presence of NA results in all values being omitted from output.

**To Reproduce**
```r
data <- data.frame(i=1:3)
data$x <- list(1, c(2, 3), c(4, NA, 5)) # The third value is going to be broken

path <- tempfile("test", fileext=".xlsx")
wb <- openxlsx::createWorkbook()
openxlsx::addWorksheet(wb, "Sheet1")
openxlsx::writeDataTable(wb, sheet=1, data, sep=";")
openxlsx::saveWorkbook(wb, file=path, overwrite=TRUE)
```

**Expected behavior**
The excel spreadsheet should contain following table:
```
i  x
1  1
2  2;3
3  4;;5
```

**Screenshots**
![image](https://user-images.githubusercontent.com/66568/142694372-b7276612-5bed-4644-81fe-93e785493ea8.png)

**Additional context**
R 4.1.1, openxlsx 4.2.4

**Comment by JanMarvin** [comment]
Hi @romanzenka , thanks for the report. I can reproduce the issue with current master. Just asking is there any reason why you expect `4;;5`? Is this documented somewhere? I wonder if we should return `2;#N/A;3` or something alike unless the user is passing something like this "na.string = ''` to writeData/writeDataTable.

The issue is the following, I'll see if I can come up with a pull request.
```R
> stringi::stri_join(c(1, NA), collapse = ";")
[1] NA
```

**Comment by JanMarvin** [comment]
fixed in #300. The following option creates the output you expect:

```R
openxlsx::writeDataTable(wb, sheet=1, data, sep=";", na.string="")
```

**Comment by romanzenka** [comment]
> Hi @romanzenka , thanks for the report. I can reproduce the issue with current master. Just asking is there any reason why you expect `4;;5`? Is this documented somewhere? I wonder if we should return `2;#N/A;3` or something alike unless the user is passing something like this "na.string = ''` to writeData/writeDataTable.

Oh I am sorry, just got this message.

Ideally this behavior is defined with the `na.string` parameter, what you did is very good. Thank you!
